### PR TITLE
Adds priority preemption support to cluster autoscaler.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

File: cluster-autoscaler/core/scale_down_test.go [hunk 1]
```diff
@@ -113,6 +124,7 @@ func TestFindUnneededNodes(t *testing.T) {
 	context := AutoscalingContext{
 		AutoscalingOptions: AutoscalingOptions{
 			ScaleDownUtilizationThreshold: 0.35,
+			ExpendablePodsPriorityCutoff:  10,
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_down_test.go]: Why is it 10? Doesn't that make every pod in test expendable (I think default is 0)?

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/scale_down_test.go]: I have added comment to flag and test case. Pods with null priority are always non expendable.

**Comment by mwielgus** [comment]
cc: @bsalamat

**Comment by mwielgus** [comment]
Thank you for preparing this pr. We will merge it as soon as we finish the main NAP effort to reduce any possible interference between new nap code and this.

File: cluster-autoscaler/core/scale_down_test.go (hunk 1 above)
Comment (bsalamat) [reply cluster-autoscaler/core/scale_down_test.go]: In order to be backward compatible, I actually think pods with no priority class or with 0 priority should be expendable. Pods with negative priority should be non-expendable by default.

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/core/scale_down_test.go]: Didn't you mean the opposite?
expendable - below 0 (or  below ExpendablePodsPriorityCutoff if defined)?
non-expendable - null priority or greater or equal to 0 (ExpendablePodsPriorityCutoff if defined)
Before priority preemption - priority was always null so all pods are non expendable.
I think I should newer check priority class. Right?

Comment (bsalamat) [reply cluster-autoscaler/core/scale_down_test.go]: Yes, reading my comment again, it should be the order way. I just edited it. Thanks and sorry for the confusion.

**Pull request merged**